client/core/trade.go

```diff
@@ -739,6 +812,67 @@ func (t *trackedTrade) redeemMatches(matches []*matchTracker) error {
 	return errs.ifany()
 }
 
+// refundMatches will send refund transactions for the specified matches.
+func (t *trackedTrade) refundMatches(matches []*matchTracker) (uint64, error) {
+	errs := newErrorSet("refundMatches: order %s - ", t.ID())
+
+	refundWallet, refundAsset := t.wallets.fromWallet, t.wallets.fromAsset // refunding to our wallet
+	var refundedQty uint64
+
+	for _, match := range matches {
+		dbMatch, _, proof, _ := match.parts()
+		if proof.RefundCoin != nil {
+			log.Errorf("attempted to execute duplicate refund for match %s, side %s, status %s",
+				match.id, dbMatch.Side, dbMatch.Status)
+			continue
+		}
+		contractToRefund := proof.Script
+		var swapCoinID order.CoinID
+		var matchFailureReason string
+		switch {
+		case dbMatch.Side == order.Maker && dbMatch.Status == order.MakerSwapCast:
```

Comment:
`isRefundable` says this is OK for maker if `dbMatch.Status == order.MakerSwapCast || (dbMatch.Status == order.TakerSwapCast && proof.MakerRedeem == nil)`
https://github.com/decred/dcrdex/pull/401/files#diff-ca928c370ec8b724badec3db0dbd7bc4R506-R507
although the reason in the second part of that `||` would be "not redeeming taker swap".  Do we not want to support that here?

Reply:
> Do we not want to support that here?

I'm guessing you mean **Do we ~not~ want to support that here?**

That's something to (re)think about.
I added a reason for doing this here in a comment a little above the referenced LOC:
```
//   The second case does not prevent Maker from (re-)attempting to redeem
//   Taker's swap; it just ensures that Maker's swap is refunded if Taker's
//   swap still can't be redeemed **after** Maker's locktime expires.
```

The idea is, if Maker is unable to redeem Taker's swap for some reason, we should allow Maker sufficient time to re-attempt the redemption. After all, Maker cannot refund their own swap until the locktime expires.
Another thing too is, by the time Maker's swap locktime expires, Taker's swap locktime would have since expired, so I think it would be a fine time to issue a refund to Maker _since_ Maker has been so far unable to redeem Taker's swap. Taker would likely have refunded their swap as well, when their own locktime expired.

Reply:
Yeah, "Do we not want to support that here?" is a (apparently colloquial) way of saying "This is not supported here, but should it be?".

Anyway, I think I understand the case, but what I'm pointing out is that this function will not allow the maker to refund in that `TakerSwapCast` case where their own maker contract is expired.  What code allows the maker to refund their swap when it is expired regardless of whether the taker's swap is broadcasted?

Reply:
Oh... I was mixing that up. It should be OK to exec the refund here as well, will fix.